addons/ai/fnc_taskDefend.sqf

```diff
@@ -25,78 +25,84 @@ Returns:
     Nil
 
 Author:
-    Rommel
+    Rommel, SilentSpike
 
 ---------------------------------------------------------------------------- */
 
-params ["_group", ["_position",[]], ["_radius",50], ["_threshold",2]];
+params ["_group", ["_position",[]], ["_radius",50,[0]], ["_threshold",2,[0]], ["_patrol",true,[true]]];
 
 _group = _group call CBA_fnc_getGroup;
 if !(local _group) exitWith {}; // Don't create waypoints on each machine
 
 _position = [_position,_group] select (_position isEqualTo []);
 _position = _position call CBA_fnc_getPos;
 
-_group enableattack false;
+[_group] call CBA_fnc_clearWaypoints;
+_group enableAttack false;
 
-private ["_count", "_list", "_list2", "_units", "_i"];
-_statics = [_position, vehicles, _radius, {(_x iskindof "StaticWeapon") && {(_x emptypositions "Gunner" > 0)}}] call CBA_fnc_getnearest;
-_buildings = _position nearObjects ["building",_radius];
-_units = units _group;
-_count = count _units;
+private _statics = _position nearObjects ["StaticWeapon", _radius];
+private _buildings = _position nearObjects ["Building", _radius];
 
+// Filter out occupied statics
+_statics = _statics select { (_x emptyPositions "Gunner") > 0 };

```

Comment:
Not compatible with the MacOS/Linux ports, currently at v1.54. Needs workaround like commy2 did recently (the "linux.pbo" addon)